packages/@aws-cdk/aws-s3objectlambda/test/s3objectlambda.test.ts

```diff
@@ -1,6 +1,666 @@
-import '@aws-cdk/assertions';
-import {} from '../lib';
+import { Template } from '@aws-cdk/assertions';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import { AccessPoint } from '../lib';
 
-test('No tests are specified for this package', () => {
-  expect(true).toBe(true);
+test('Can create a valid access point', () => {
+  const stack = new cdk.Stack();
+  const bucket = new s3.Bucket(stack, 'MyBucket');
+  const handler = new lambda.Function(stack, 'MyFunction', {
+    runtime: lambda.Runtime.NODEJS_14_X,
+    handler: 'index.hello',
+    code: new lambda.InlineCode('def hello(): pass'),
+  });
+  const accessPoint = new AccessPoint(stack, 'MyObjectLambda', {
+    bucket,
+    handler,
+    accessPointName: 'obj-lambda',
+    supportsGetObjectRange: true,
+    supportsGetObjectPartNumber: true,
+    payload: { foo: 10 },
+  });
+  new cdk.CfnOutput(stack, 'AccessPointName', {
+    value: accessPoint.accessPointName,
+  });
+  new cdk.CfnOutput(stack, 'DomainName', {
+    value: accessPoint.domainName,
+  });
+  new cdk.CfnOutput(stack, 'RegionalDomainName', {
+    value: accessPoint.regionalDomainName,
+  });
+  new cdk.CfnOutput(stack, 'VirtualHostedUrl', {
+    value: accessPoint.virtualHostedUrlForObject('key', {
+      regional: true,
+    }),
+  });
+  new cdk.CfnOutput(stack, 'VirtualHostedRegionalUrl', {
+    value: accessPoint.virtualHostedUrlForObject('key', {
+      regional: false,
+    }),
+  });
+  expect(Template.fromStack(stack).toJSON()).toEqual({
```

Comment:
Can you update these tests to just use the `assertions` methods like

```ts
Template.fromStack(stack).hasResourceProperties('AWS::S3ObjectLambda::AccessPoint', {
  ObjectLambdaConfiguration: {},
});
```

Sorry if my comment about using `expect(Template.fromStack(stack).toJSON())` was misleading. I was meaning that that method can be useful to find out logical ids of resources if the normal test output isn't giving you what you need.

Also to make things easier you might want to just create the resources in a `beforeEach` and then have a separate test to assert individual things. Combined with using the `assertions` methods each test can be smaller and validate a smaller piece of the template, which makes it easier to review.

```ts
let stack: cdk.Stack;
let bucket: s3.Bucket;
let handler: lambda.Function;
beforeEach(() => {
  stack = new cdk.Stack();
  bucket = new s3.Bucket(stack, 'MyBucket');
  handler = new lambda.Function(stack, 'MyFunction', {
    runtime: lambda.Runtime.NODEJS_14_X,
    handler: 'index.hello',
    code: new lambda.InlineCode('def hello(): pass'),
  });
});

test('can create a valid access point');
test('regional virtual hosted URL');
test('non-regional virtual hosted URL');
...etc.

Reply:
Updated!